app/models/manageiq/providers/ansible_tower/shared/automation_manager/credential.rb

```diff
@@ -2,7 +2,15 @@ module ManageIQ::Providers::AnsibleTower::Shared::AutomationManager::Credential
   extend ActiveSupport::Concern
 
   module ClassMethods
+    def provider_params(params)
+      params[:username] = params.delete(:userid) if params.include?(:userid)
+      params[:username] = params.delete('userid') if params.include?('userid')
```

Comment:
string type is coming from API calls via HTTP.

You want me to take away the line handling `:userid`?  I keep it there so it will be consistent with all other attributes as in [here](https://github.com/jameswnl/manageiq/blob/f0245a25e287df552c34615c19773e7c24d0ffdc/app/models/manageiq/providers/ansible_tower/shared/automation_manager/machine_credential.rb#L1-L1)

Other usage through ruby method invocation can use whole set of attributes using `symbol` key.

or else will have to change all the credentials' `ATTRIBUTES` hash keys.